### ISSUE TITLE
[ban-types] Support for Generic Arrow Functions in .tsx files

### ISSUE THREAD

**Issue by dimitropoulos** [typescript-eslint/typescript-eslint#2191]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-types": "error"
  }
}
```

```TS
// my-file.tsx
const myHandler = <T extends {} = HTMLElement>(event: ChangeEvent<T>) => {
    ...
}
```

**Expected Result**
No error is reported because this usage of `{}` is required by TypeScript in this situation as a kind of ["forever-hack"](https://github.com/Microsoft/TypeScript/issues/4922#issuecomment-142383182).

I still want usages of `{}` to be disallowed as normal.  Just this one edge case for generic arrow functions and `extends` should be allowed.

**Actual Result**
```
Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
eslint@typescript-eslint/ban-types
```
is reported.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.1.0` |
| `@typescript-eslint/parser`        | `3.1.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `12` |
| `npm`                              | `12` |

**Comment by bradzacher** [comment]
I believe you can use `extends unknown` instead. That would save this from happening, and would be clearer. To me, if I saw `extends {}` I would think your intention was to enforce the generic is an object type, but if I saw `extends unknown` or `extends any`, then I would immediately recognise it was there for another reason.

**Comment by dimitropoulos** [comment]
I get that... but.. well.  Ok.  I've been using `extends {}` as the "approved workaround" for quite some time now - predating `unknown`'s existence (now that I think about it).

I'll mention that it's still pretty common ([TypeScript Deep Dive suggests it](https://basarat.gitbook.io/typescript/tsx/react#generic-functions)), but at least if someone else has this problem in the future they can come back to this issue.

I'll just change my codebases and my habit to use `extends unknown` and report back (perhaps with a separate issue) if anything more is required on this topic.
